tedge_config/src/config_setting.rs

```diff
@@ -0,0 +1,42 @@
+pub trait ConfigSetting {
+    /// This is something like `device.id`.
+    const KEY: &'static str;
+
+    const DESCRIPTION: &'static str;
```

Comment:
It would be good to add a comment telling this description is used for `tedge config list`.